proto/redpanda/api/dataplane/v1alpha1/acl.proto

```diff
@@ -0,0 +1,182 @@
+}
+
+message ListACLsResponse {
+  message Policy {
+    string principal = 1;
+    string host = 2;
+    ACL.Operation operation = 3;
+    ACL.PermissionType permission_type = 4;
+  }
+
+  ACL.ResourceType resource_type = 1;
+  string resource_name = 2;
+  ACL.ResourcePatternType resource_pattern_type = 3;
+  repeated Policy acls = 4;
+}
+
+message CreateACLRequest {
+  ACL.ResourceType resource_type = 1;
+  string resource_name = 2;
+  ACL.ResourcePatternType resource_pattern_type = 3;
+  string principal = 4;
+  string host = 5;
+  ACL.Operation operation = 6;
+  ACL.PermissionType permission_type = 7;
+}
+
+message CreateACLResponse {}
+
+message DeleteACLsRequest {
+  message Filter {
+    ACL.ResourceType resource_type = 1;
+    string resource_name = 2;
+    ACL.ResourcePatternType resource_pattern_type = 3;
+    string principal = 4;
+    string host = 5;
+    ACL.Operation operation = 6;
+    ACL.PermissionType permission_type = 7;
+  }
+
+  Filter filter = 1;
+}
+
+message DeleteACLsResponse {
+  message MatchingACL {
+    ACL.ResourceType resource_type = 1;
+    string resource_name = 2;
+    ACL.ResourcePatternType resource_pattern_type = 3;
+    string principal = 4;
+    string host = 5;
+    ACL.Operation operation = 6;
+    ACL.PermissionType permission_type = 7;
+    int32 error_code = 8;
+    string error_message = 9;
+  }
+
+  repeated MatchingACL matching_acls = 1;
+}
+
+service ACLService {
+  rpc ListACLs(ListACLsRequest) returns (ListACLsResponse) {
+    option (google.api.http) = {get: "/v1alpha1/acls"};
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "List ACLs"
+      description: "List ACLs"
+      responses: {
+        key: "200"
+        value: {
+          description: "OK"
+          schema: {
+            json_schema: {ref: ".redpanda.api.dataplane.v1alpha1.ListACLsResponse"}
+          }
+        }
+      }
+    };
+  }
+
+  rpc CreateACL(CreateACLRequest) returns (CreateACLResponse) {
+    option (google.api.http) = {post: "/v1alpha1/acls"};
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Create ACL"
+      description: "Create ACL"
+      responses: {
+        key: "201"
+        value: {
+          description: "OK"
+          schema: {
+            json_schema: {ref: ".redpanda.api.dataplane.v1alpha1.CreateACLResponse"}
+          }
+        }
+      }
+    };
+  }
+
+  rpc DeleteACLs(DeleteACLsRequest) returns (DeleteACLsResponse) {
+    option (google.api.http) = {delete: "/v1alpha1/acls"};
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Delete ACLs"
+      description: "Delete all ACLs that match the given filter"
+      responses: {
+        key: "200"
+        value: {
+          description: "OK"
+          schema: {
+            json_schema: {ref: ".redpanda.api.dataplane.v1alpha1.DeleteACLsResponse"}
+          }
+        }
+      }
+    };
+  }
+}
```

Comment:
We only drafted all the protos because Alex requested an OpenAPI spec. We'll work on the actual implementation including validation, better comments etc. in a later follow-up